text/0000-const-type-inference.md

```diff
@@ -0,0 +1,114 @@
+- Feature Name: const_type_inference
+- Start Date: 2023-12-21
+- RFC PR: [rust-lang/rfcs#3546](https://github.com/rust-lang/rfcs/pull/3546)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Allow type inference for `const` or `static` when the type of the initial value is known.
+
+# Motivation
+[motivation]: #motivation
+
+Rust currently requires explicit type annotations for `const` and `static` items.
+
+
+In simple cases, explicitly writing out
+the type of the const seems trivial. However, this isn't always the case:
+
+- Sometimes the constant's value is complex, making the explicit type overly verbose.
+- In some cases, the type may be unnameable.
+- When creating macros, the precise type might not be known to the macro author.
+- Code generators may not have enough information to easily determine the type.
+
+This change aims to make Rust code more concise and maintainable, especially in scenarios where the types of
+const items are complicated or not easily expressible.
+
```

Comment:
It might also be helpful to add ergonomics to the motivation. This particularly impacts new Rust users coming from languages where most or all types are inferred.

Here is a draft paragraph:
```suggestion

Inferring constant types also improves the ergonomics of the language, particularly for new users. Some users are
coming from languages where most (or all) types are inferred. So inferring obvious types matches their
expectations. Other new users are focused on learning ownership, or other core Rust concepts. Reducing the
amount of boilerplate reduces their mental load. This reduction in mental load also helps experienced programmers.

```

Reply:
Thank you!